File: Samples/NumberOfPages/C#/NumberOfPages.cs
using System;
using System.Text;
using System.Windows.Forms;

using BitMiracle.LibTiff.Classic;

namespace BitMiracle.LibTiff.Samples
{
    public static class NumberOfPages
    {
        public static void Main()
        {
            const string fileName = "Sample Data/multipage.tif";

            using (Tiff image = Tiff.Open(fileName, "r"))
            {
                if (image == null)
                {
                    MessageBox.Show("Could not open incoming image");
                    return;
                }

                StringBuilder message = new StringBuilder();
                message.AppendFormat("Tiff.NumberOfDirectories() returns {0} pages\n", image.NumberOfDirectories());
                message.AppendFormat("Enumerated {0} pages", CalculatePageNumber(image));
                
                MessageBox.Show(message.ToString());
            }
        }

        private static int CalculatePageNumber(Tiff image)
        {
            int pageCount = 0;
            do
            {
                ++pageCount;
            } while (image.ReadDirectory());

            return pageCount;
        }
    }
}

File: Samples/NumberOfPages/VB.NET/NumberOfPages.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Text
Imports System.Windows.Forms

Imports BitMiracle.LibTiff.Classic

Namespace BitMiracle.LibTiff.Samples
    Public NotInheritable Class NumberOfPages
        Private Sub New()
        End Sub
        Public Shared Sub Main()
            Const fileName As String = "Sample Data/multipage.tif"

            Using image As Tiff = Tiff.Open(fileName, "r")
                If image Is Nothing Then
                    MessageBox.Show("Could not open incoming image")
                    Return
                End If

                Dim message As New StringBuilder()
                message.AppendFormat("Tiff.NumberOfDirectories() returns {0} pages" & vbCrLf, image.NumberOfDirectories())
                message.AppendFormat("Enumerated {0} pages", CalculatePageNumber(image))

                MessageBox.Show(message.ToString())
            End Using
        End Sub

        Private Shared Function CalculatePageNumber(ByVal image As Tiff) As Integer
            Dim pageCount As Integer = 0
            Do
                pageCount += 1
            Loop While image.ReadDirectory()

            Return pageCount
        End Function
    End Class
End Namespace